### PR TITLE
[76] [UI] As a logged in user, I can see a different answer view type

### LINKED ISSUE #76: [UI] As a logged in user, I can see a different answer view type
## Why

- We need an answer view to allow the user to complete the survey.
 
## Acceptance Criteria

- Show answer (with different type)
- The user able to interact with each type of answer

Screen Shot 2021-06-21 at 6 07 16 PM
Screen Shot 2021-06-21 at 6 07 30 PM

### PR DESCRIPTION
**Pull request by chiiida**
https://github.com/llleyelll/ic-surveys-ios/issues/76

## What happened 👀

- Implement answers for each question type
 
## Insight 📝

- This PR was broken down from [Question Screen UI PR](https://github.com/llleyelll/ic-surveys-ios/pull/75)
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/45258998/124743145-a0e58300-df47-11eb-8e08-0418f1bc1b02.mov

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: SurveyApp/Extensions/UIKit/UIImage/UIImage+Creation.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        let rect = CGRect(origin: CGPoint(x: 0, y: 0), size: size)
+        UIGraphicsBeginImageContext(size)
+        let context = UIGraphicsGetCurrentContext()
+        context?.setFillColor(color.cgColor)
+        context?.fill(rect)
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return image
+    }
+}
```
Comment (blyscuit) [inline SurveyApp/Extensions/UIKit/UIImage/UIImage+Creation.swift]:
```suggestion
extension UIImage {

    static func withBackground(color: UIColor, size: CGSize = CGSize(width: 1, height: 1)) -> UIImage? {
        let renderer = UIGraphicsImageRenderer(size: size)
        return renderer.image { ctx in
            
    color.setFill()
    ctx.fill(CGRect(x: 0, y: 0, width: size.width, height: size.height))
            self.drawHierarchy(in: self.bounds, afterScreenUpdates: true)
        }
    }
}
```

Please use `UIGraphicsImageRenderer` as it is faster https://medium.com/the-traveled-ios-developers-guide/uigraphicsimagerenderer-fe40edc3a464

Please check if this code is working as expected or not.

File: SurveyApp/Views/AnswerView/DisplayType.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
+//
+//  DisplayType.swift
+//  SurveyApp
+//
+//  Created by Chananchida F. on 7/6/21.
+//
+
+import UIKit
+
+enum DisplayType {
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/DisplayType.swift]: This is a global variable, let's give it a more explanatory name `AnswerDisplayType` or something else?

File: SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    private func setUpLayout() {
+        addSubview(tableView)
+        
+        tableView.snp.makeConstraints {
+            $0.centerY.equalToSuperview()
+            $0.leading.equalToSuperview().inset(10.0)
+            $0.trailing.equalToSuperview().inset(20.0)
+            $0.height.equalTo(300.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift]: what will happen to smaller screen? (iPhone SE). I suggest make this height constraint to parent's aspect ratio and make it vertically scrollable (if not already)?

File: SurveyApp/Views/AnswerView/NPSAnswerView.swift [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    private func setUpLayout() {
+        addSubview(segmentedControl)
+        addSubview(mostLikelyLabel)
+        addSubview(leastLikelyLabel)
+
+        segmentedControl.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.width.equalTo(UIScreen.main.bounds.width - 40.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/NPSAnswerView.swift]: can this be constraint to parent?

**Review by blyscuit — CHANGES_REQUESTED** [review]

File: SurveyApp/Views/AnswerView/TextFieldAnswerView.swift [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setUpLayout() {
+        addSubview(textView)
+        addSubview(stackView)
+
+        textView.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.width.equalTo(UIScreen.main.bounds.width - 40.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/TextFieldAnswerView.swift]: sorry I missed this one. Can this also be constraint to parent?

File: SurveyApp/Views/AnswerView/TextFieldAnswerView.swift [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            $0.height.equalTo(168.0)
+        }
+
+        stackView.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.width.equalTo(UIScreen.main.bounds.width - 40.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/TextFieldAnswerView.swift]: sorry I missed this one.

File: SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    private func setUpLayout() {
+        addSubview(tableView)
+        
+        tableView.snp.makeConstraints {
+            $0.leading.equalToSuperview().inset(10.0)
+            $0.trailing.equalToSuperview().inset(20.0)
+            $0.top.equalToSuperview().inset(50.0)
+            $0.height.equalToSuperview().inset(50.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift]: if you want maximum height 300 you can do
```suggestion
            $0.height.lessThanOrEqualTo(300.0)
            $0.height.lessThanOrEqualToSuperview().inset(50.0)
```

File: SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift [hunk 8]
```diff
@@ -32,7 +32,7 @@ class MultipleChoiceAnswerView: UIView {
             $0.leading.equalToSuperview().inset(10.0)
             $0.trailing.equalToSuperview().inset(20.0)
             $0.top.equalToSuperview().inset(50.0)
-            $0.height.equalToSuperview().inset(50.0)
+            $0.height.lessThanOrEqualTo(300.0)
```
Comment (blyscuit) [inline SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: Can you check if you need to add `$0.height.lessThanOrEqualToSuperview().inset(50.0)` for smaller device.

Comment (chiiida) [reply SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: @blyscuit  I tested `$0.height.lessThanOrEqualToSuperview().inset(50.0)` with iPhone SE but the table view doesn't display.

This is result when use `$0.height.lessThanOrEqualTo(300.0)` on iPhone SE:

Screen Shot 2021-07-08 at 6 10 30 PM

Comment (blyscuit) [reply SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: you are right, it should've been `$0.bottom.lessThanOrEqualToSuperview().inset(20.0)`

Comment (chiiida) [reply SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: @blyscuit Sorry, I misunderstood.
It's looks better when add both `$0.height.lessThanOrEqualTo(300.0)` and `$0.height.lessThanOrEqualToSuperview().inset(50.0) `:
I will update the code, please help check again. Thank you 🙏

Screen Shot 2021-07-08 at 6 21 02 PM

Comment (chiiida) [reply SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: > you are right, it should've been `$0.bottom.lessThanOrEqualToSuperview().inset(20.0)`

@blyscuit  This approach got the same result and work fine on both small devices and large devices. I updated the code to this approach instead. 
Please help check. Thank you so much🙇‍♀️

**Review by blyscuit — APPROVED** [review]

File: SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift (hunk 8 above)
Comment (blyscuit) [reply SurveyApp/Views/AnswerView/MultipleChoiceAnswerView.swift:36]: I mean, it should've been the same right, I expect small device will show smaller table 🤔 This is from iPod 7.

**Pull request merged**